Mental Conditions / Give Psych a Purpose

Kyep opened the pull request:
Based on:
- http://nanotrasen.se/phpBB3/viewtopic.php?f=12&t=7958
- http://nanotrasen.se/forum/viewtopic.php?f=2&t=8003

Goals:
- Give the psychiatrist something meaningful to do
- Create a new sort of "hard mode" for people wishing to play challenged characters

Changes:
- Adds 4 mental illnesses to the game: Psychosis, Eating Disorder, Depression, and Anxiety.
- These are treated as permanent disabilities: there is little in the way of treatment for them, and the only way to get them is to opt-in to them during character setup, in the [Set Disabilities] section.
- Psychosis simply means you have hallucinations / lsd-like experiences sometimes. It doesn't give you any special permission to be violent - and the description on the preferences screen notes this.
- Eating Disorder means you underestimate how much food you need, and can starve without proper nutrition management.
- Depression means you move slowly, see the world in black-and-white, and rarely have a chance to suicide if you are holding a gun.
- Anxiety will randomly make you jittery/dizzy. Rarely, you can have a full-blown attack which stuns you for a short time.
- All of these conditions (plus: nearsightedness, obesity, epilepsy, deafness, blindness, and muteness) now show up as major disabilities in medical records. Yes, medical records actually have a use now. Your mental status gets set to *WATCH* if you have one of these conditions.
- Mental conditions also disqualify you from serving in any command or security job. So do the physical disabilities deafness, blindness, and muteness. A head of staff who can't hear, see or talk isn't much use.
- Most of these new mental traits will be revealed if someone examines you. Masks may prevent this for some of them.
- Psychosis is treated with the psych's anti-hallucinogenic drugs. Eating Disorder requires proper nutrition management to avoid starvation. Depression is untreatable but can be managed simply by not touching guns. Anxiety is also untreatable but the effects are generally limited to jitteriness/dizzyness, with the occasional stun. 

Note:
- This is an experimental PR. If people aren't keen on it, I'll close it. Simply putting it up here because some people said they'd really like to see it.
- This is still a work in progress. Ideally I'd like to add more direct tie-ins to the psych, but I also wanted to get a bit more feedback (and find out how much people like the concept) before I spend more time coding it.

FreeStylaLT commented:
Is there a way to treat these?

FlattestGuitar commented:
I might get branded the PC warrior here, but people with depression randomly suiciding when holding a gun seems pretty ignorant and offensive to me. It's just a vidya game, but still...

Kyep commented:
@FreeStylaLT They're not curable, but you can mitigate their effects using the techniques described in the post. I'm debating adding more treatment options, but before doing any further work on this I felt I had to put up a PR and see if it was worth pursuing or not.

@FlattestGuitar The real equivalents of these conditions are always going to be way, way more complex than I can model with code. Whatever I put in is always going to be a drastic simplification. Feel free to suggest better models.

FlattestGuitar commented:
A better model would be leaving mental disabilities to RP, just like it always has been. 

I honestly don't see the problem with people RPing a mental disability, this PR seems redundant to me.

IcyV commented:
:-1: Space is generally supposed to fulfill a purpose of interesting and fun rounds. Psychiatric conditions horribly based on their real counterparts is something that should be left to RP should someone choose to do so. I'd rather see creative conditions that fit the theme of wackiness and craziness that is SS13 instead of real issues.

CrAzYPiLoT-SS13 commented:
Space Madness: The condition which appears when a patient is confined in a
spacecraft with similarly decorated surroundings and an always-stressful
atmosphere for prolonged periods of time.

Fox-McCloud commented:
This doesn't give the psych something to do. This gives players a way to grief the psych and CMO because they have a "disorder".

Only you're going to end up wasting your entire shift "treating" them (when it's not even possible), then have that very player complain to no end about the psych/CMO not "doing their job" because they refuse to deal with something that's (1) untreatable (2) largely going to re-inforce the medbay attention-seekers who self-harm/make life difficult for medbay PURELY to suck up resources and hog attention.

I'm going to echo some of the things @IcyV said and finish off with what @FlattestGuitar said with regard to RP.

Also individuals with depression aren't necessarily suicidal--everyone handles that differently.

Aurorablade commented:
I like this pr I do rp a crazy person and this seems like a good way to fluff it up. And trust me even without this people would find a way to grief medical so that is a moot point IMO. As someone with depression I am not offended by its implantation here but I can see how someone could be.    

If anything we could try to prsent alternatives for conditions such as with space madness

Another thing to conser is maybe useing like a point based system for disabiltys just to balance out medical griff. Even if it is funny to give a char all the disabilities at once

FalseIncarnate commented:
Haloperidol could feasibly treat the psychosis symptoms (it counters druggy and hallucination effects), but the others seem rather untreatable in any sort of meaningful way. Of the listed disabilities, only psychosis and anxiety can really be off-set through pure medication, and are the least likely to result in requiring additional medical attention. The eating disorder actually does organ damage if you don't eat enough, leading to lots of surgery or mitocholide being spent on someone (arguably the biggest waste of medical's time and resources), and frankly that seems WAY too powerful for something like an eating disorder, which would have to be incredibly severe to result in self-starvation, not to mention eating disorders also covers the opposite end of the spectrum where you eat too much.

Fox-McCloud commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -158,6 +158,66 @@
+				if (prob(50))
+					suicide()
+				else
+					to_chat(src, "[src] is looking appealing...")
```
"Bobby is looking appealing..." is what this will result in.

Fox-McCloud commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -158,6 +158,66 @@
+			else
+				var/obj/item/organ/internal/lungs/O = get_int_organ(/obj/item/organ/internal/lungs)
+				if(O)
+					O.damage += 1
```
This has no safeties and will blow right through and organ's max damage.

Regen1 commented:
I am a fan of the idea, however I hate the execution.

This just uses "old" methods of affecting the player, dizziness, confusion and all that.

I would be much more interested if this added more unique effects to the game, a good example is Lifeweb's "Freaking out", where the screen shakes a lot and adds a blurry filter over, as if the character is freaking out. It makes it harder, but not impossible, to click things and navigate.

Da-Dman234 commented:
"Regen1"
> I am a fan of the idea, however I hate the execution.
This just uses "old" methods of affecting the player, dizziness, confusion and all that.
I would be much more interested if this added more unique effects to the game, a good example is Lifeweb's "Freaking out", where the screen shakes a lot and adds a blurry filter over, as if the character is freaking out. It makes it harder, but not impossible, to click things and navigate.

This. I still really like the idea, but the execution/available disorders really aren't cutting it for me. I also kinda have a problem with the depression thing simply because Depressed =/= Suicidal.

Earthdivine commented:
Please do not connect depression and suicide. A better form of it would be to add slowdown to mobs or longer length of time to construct things. Doing it this way is very poor form.

IcyV commented:
I'd honestly love to see SS13 themed conditions rather than real ones. A mental state that makes people "clowny" and makes random things look like bananas, for instance. Silly or more properly fitted things instead of genuine issues.

Deanthelis commented:
I was hoping this would be a PR with mental diseases you'd get from certain behaviors in-game. Maybe Engineers wearing Hardsuits all shift would get some sort of agoraphobia if they take it off based on how many open tiles are around them, or a compulsion to stare at the singularity the more you stare at the singularity, or funny stuff like that, some of which could be used to curb certain silly behaviors (like wearing a hardsuit in the station all shift long).

Instead it's just a blatant display of ignorance, and not at all the memes I was hoping for. And to think, you could've named one of them Honkington's Disease.

Kyep commented:
Update 3 has:
- Removed all organ damage from eating disorder, and converted it to much, much less lethal brute damage.
- Removed the auto-suicide chance from depression.

The remaining objections seem to be: 
1) Leave disabilities to RP - which seems a little odd since we already have physical disabilities in the game, and the exact same argument could be made for those, but nobody advocates for the removal of physical conditions like blindness or deafness. If its OK to have physical conditions, why not mental ones?
2) Would lead to people griefing medical by having them - which I don't think is true. Anyone who wants to grief already has tons of options to do so, not least the physical disabilities. Try conversing with someone who is blind, deaf and mute. Just try it.
3) Use brand-new effects like lifeweb's "freaking out" - some problems with this one. First is I don't play lifeweb, nor do I want to play it, and its wiki is terrible, so there's very little info about what sort of effects you might mean. Second is that I'm not sure how to create them. Third is, in the absence of information, even if I did go back to the drawing board and create new effects, I'm not sure if they'd be sufficiently similar to what you're thinking of that they'd be what you want.
4) Use SS13-themed conditions instead: SS13-themed conditions could be interesting, but I don't want to make them silly like 'clownish disorder' or similar, as I feel that's cheap and ridiculous. 

The last update has resolved the two biggest objections to this PR - but I'm still going to close it. 
Aside from the votes being against it, the remaining objections seem so broad that they'd apply to any PR like this, no matter the implementation. So, it looks like changing the code won't sway opinion on it.

Aurorablade commented:
i disagree and feel this PR should be reopened and continued...

Kyep commented:
@Aurorablade If you'd like to take my code and extend/revise it, you're welcome to do so.
